nft_ingester/src/backfiller.rs

```diff
@@ -0,0 +1,702 @@
+                        println!("Transaction has error: {err}");
+                        continue;
+                    }
+                    meta
+                } else {
+                    println!("Unexpected, EncodedTransactionWithStatusMeta struct has no metadata");
+                    continue;
+                };
+
+                // Get `UiTransaction` out of `EncodedTransactionWithStatusMeta`.
+                let ui_transaction = match tx.transaction {
+                    solana_transaction_status::EncodedTransaction::Json(ref ui_transaction) => {
+                        ui_transaction
+                    }
+                    _ => {
+                        return Err(IngesterError::RpcDataUnsupportedFormat(
+                            "EncodedTransaction".to_string(),
+                        ));
+                    }
+                };
+
+                // See if transaction is a vote.
+                let ui_raw_message = match &ui_transaction.message {
+                    solana_transaction_status::UiMessage::Raw(ui_raw_message) => {
+                        if ui_raw_message.account_keys.iter().any(|key| key == VOTE) {
+                            // Debug.
+                            println!("Skipping vote transaction");
+                            continue;
+                        } else {
+                            ui_raw_message
+                        }
+                    }
+                    _ => {
+                        return Err(IngesterError::RpcDataUnsupportedFormat(
+                            "UiMessage".to_string(),
+                        ));
+                    }
+                };
+
+                // Filter out transactions that don't have to do with the tree we are interested in or
+                // the Bubblegum program.
+                let tree = bs58::encode(tree).into_string();
+                let bubblegum = BubblegumProgramID().to_string();
+                if ui_raw_message
+                    .account_keys
+                    .iter()
+                    .all(|pk| *pk != tree && *pk != bubblegum)
+                {
+                    // Debug.
+                    println!("Skipping tx unrelated to tree or bubblegum PID");
+                    continue;
+                }
+
+                // Debug.
+                println!("Serializing transaction");
+                // Serialize data.
+                let builder = FlatBufferBuilder::new();
+                let builder = serialize_transaction(
+                    builder,
+                    &meta,
+                    &ui_raw_message,
+                    slot.try_into().unwrap(),
+                )?;
+
+                // Debug.
+                println!("Putting data into Redis");
+                // Put data into Redis.
+                let _ = self
+                    .messenger
+                    .send(TRANSACTION_STREAM, builder.finished_data())
```

Comment:
Yeah once I had a `UiTransaction` and it contained a `UiMessage` and I recognize all the same data from `ReplicaTransactionInfo`, it made sense to me to go this route rather than re-write or modify code to order instructions, parse logs, etc.

I just can seriallize it and let the working program parsers you wrote do the consumer work!

One interesting thing this causes is when the normal program parsers operate, they insert more rows in the `backfill_items` table, but this does not cause problems because I use the `backfilled` flag to not care about entries below the sequence number up to which I have already backfilled everything.